content/sensu-go/6.3/observability-pipeline/observe-schedule/rule-templates.md

```diff
@@ -0,0 +1,471 @@
+---
+title: "Rule templates reference"
+linkTitle: "Rule Templates Reference"
+reference_title: "Rule templates"
+type: "reference"
+description: "Sensu's rule templates are resources that Sensu applies to business service components. The rules define the conditions under which Sensu will consider a service component online, degraded, or offline. Read this reference to learn more about Sensu's rule templates for business service management."
+weight: 70
+version: "6.3"
+product: "Sensu Go"
+menu: 
+  sensu-go-6.3:
+    parent: observe-schedule
+---
+
+**COMMERCIAL FEATURE**: Access business service management, including rule templates, in the packaged Sensu Go distribution.
+For more information, see [Get started with commercial features][4].
+
+Rule templates are resources that Sensu applies to business service [components][3].
+The rules define the conditions under which Sensu will consider a service component online, degraded, or offline.
+Rules apply to aggregate data derived from a service component’s selection of Sensu Go events.
+
+In rule templates, you configure the conditions for creating an event with an ECMAScript 5 (JavaScript) expression.
+When you add rule templates by reference in your service component definitions, Sensu evaluate events according to the monitoring rules you configure in the rule template.
+Monitoring rules create events when one or more thresholds are exceeded; for example, a threshold might be fewer than 70% of a service component's selected events have a check status of OK.
+
+Components can reference more than one rule template.
+Sensu evaluates each rule separately, and each rule produces its own event as output.
+
+## Rule template example
+
+This example rule template creates an event when the percentage of events with the given status exceed the given threshold:
+
+{{< language-toggle >}}
+
+{{< code yml >}}
+---
+type: RuleTemplate
+api_version: bsm/v1
+metadata:
+  name: status-threshold
+  namespace: default
+  created_by: admin
+spec:
+  description: Creates an event when the percentage of events with the given status exceed the given threshold
+  arguments:
+    required:
+      - threshold
+    properties:
+      status:
+        type: string
+        default: non-zero
+        enum:
+          - non-zero
+          - warning
+          - critical
+          - unknown
+      threshold:
+        type: number
+        description: Numeric value that triggers an event when surpassed
+  eval: |
+    var statusMap = {
+      "non-zero": 1,
+      "warning": 1,
+      "critical": 2,
+    };
+    function main(args) {
+      var total = sensu.events.count();
+      var num = sensu.events.count(args.status);
+      if (num / total <= args.threshold) {
+        return;
+      }
+      return sensu.new_event({
```

Comment:
We have removed the `sensu.NewEvent()` API ; one needs to directly mutate the `event` object representing the event that's going to be emitted by the rule. The following should do the trick:

```
event.status = statusMap[args.status]
```